modules/core/src/main/java/org/apache/ignite/internal/processors/metric/MetricRegistry.java

```diff
@@ -0,0 +1,196 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.internal.processors.metric;
+
+import java.util.function.BooleanSupplier;
+import java.util.function.DoubleSupplier;
+import java.util.function.IntSupplier;
+import java.util.function.LongSupplier;
+import java.util.function.Supplier;
+import org.apache.ignite.internal.processors.metric.impl.DoubleMetricImpl;
```

Comment:
What's the point in having Impl suffix ?
org.apache.ignite.internal.processors.metric.impl.DoubleMetric looks cleaner.

Reply:
`DoubleMetric` is the name of the interface. You can find it [here](https://github.com/apache/ignite/pull/6546/files#diff-2f22f348ff33397a422d8b9d1a766c13). I think we should avoid classes with the same names but different packages.